### DISCUSSION TITLE
High correlation among topics after HDSCAN

### DISCUSSION THREAD

**Discussion by chanansh** [MaartenGr/BERTopic#2111] [Q&A]
I've used the following settings:

```
embedding_model = SentenceTransformer("all-MiniLM-L6-v2") # sentence transformer model
device = torch.device("mps") if torch.backends.mps.is_available() else torch.device("cpu") # detect if mps is available
embedding_model.to(device) # send model to device
vectorizer_model = CountVectorizer(stop_words="english", min_df=1, ngram_range=(1, 2)) # vectorizer model   
keybert_model = KeyBERTInspired() # representation model
representation_model = {"KeyBERT": keybert_model} # representation model
umap_model = UMAP(n_neighbors=15, n_components=5, min_dist=0.0, metric='cosine', random_state=42) # dimension reduction
hdbscan_model = HDBSCAN(min_cluster_size=10, metric='euclidean', cluster_selection_method='eom', prediction_data=True) # clustering

topic_model = BERTopic(
  # Pipeline models
  embedding_model=embedding_model,
  umap_model=umap_model,
  hdbscan_model=hdbscan_model,
  vectorizer_model=vectorizer_model,
  representation_model=representation_model,

  # Hyperparameters
  top_n_words=10,
  verbose=True
)

```
However, when running similarity matrix after the modeling, I am still left with VERY (>0.95) correlated topics. 
What am I doing wrong?
![image](https://github.com/user-attachments/assets/134ead40-df6c-463f-be67-37e3935eabe1)

**Comment by MaartenGr** [comment]
There isn't something wrong with your output. You are comparing the topic embeddings, which are compressed representations of the entire cluster, as a proxy for the cluster itself to do the similarity mapping on. It is not unexpected that the centroid of two clusters are at the same location since clusters can have different shapes. So just because two topic embeddings have a high similarity (not correlation!) doesn't mean that the cluster model found two of the exact same clusters.

**Comment by chanansh** [comment]
Well, according to the content they are pretty much the same

Thanks,
HS
________________________________

**Reply by MaartenGr** [reply]
Note that this depends highly on a number of factors, including the embedding model, dimensionality reduction model, and cluster model. For instance, the embedding model might still see some differences in the content which is captured in the dim/cluster models.

If low similarity is important (which again, does not necessarily relate to the quality of clusters), then you can use k-Means instead of HDBSCAN which operates directly on centroids and attempts to space them relatively far apart. You could also fine-tune BERTopic with `nr_topics="auto"` to automatically merge topics, or even use the `.merge_topics` function BERTopic has to manually merge the topics with high similarity. Fortunately, there are many methods you can use.
